### PR TITLE
Certificate and VM hostname validation improvements

### PR DESCRIPTION
**Pull request by sureshanaparti**
### Description

This PR improves the Certificate and VM hostname validations, during direct download certificate setup and VM IP address check.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

Tested upload templatedirectdownloadcertificate with valid and invalid names.

### CODE REVIEW

**Comment by sureshanaparti** [comment]
@blueorangutan package

File: utils/src/main/java/com/cloud/utils/net/NetUtils.java [hunk 1]
```diff
@@ -99,6 +99,10 @@ public class NetUtils {
     public final static int IPV6_EUI64_11TH_BYTE = -1;
     public final static int IPV6_EUI64_12TH_BYTE = -2;
 
+    // Regex
+    public final static Pattern HOSTNAME_PATTERN = Pattern.compile("[a-zA-Z0-9-]+");
+    public final static Pattern START_HOSTNAME_PATTERN = Pattern.compile("^[0-9-].*");
```
Comment (DaanHoogland) [inline utils/src/main/java/com/cloud/utils/net/NetUtils.java:104]: would this work?

```suggestion
    public final static Pattern HOSTNAME_PATTERN = Pattern.compile("^(a-zA-Z)[a-zA-Z0-9-]+");
```

we could skip the START_HOSTNAME_PATTERN check if we check the hot name like this

Comment (sureshanaparti) [reply utils/src/main/java/com/cloud/utils/net/NetUtils.java:104]: yes, can skip the start pattern for host name. for now, keeping it as separate check.

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Review by vishesh92 — APPROVED** [review]
clgtm. didn't test.

**Review by kiranchavala — APPROVED** [review]
LGTM

Executed the API “uploadtemplatedirectdownloadcertificate" by providing an invalid name value

https://cloudstack.apache.org/api/apidocs-4.18/apis/listTemplateDirectDownloadCertificates.html

For example 
Invalid name with 
greater than 63 chars
should not start or end with '-' 
should only contain alpha numeric

example: 

```
(localcloud) 🐱 > upload templatedirectdownloadcertificate hypervisor=KVM name=-kiran certificate="MIIF7zCCA9egAwIBAgIUcC+POP+CQ6CUMfeMJqelQO/zSxAwDQYJKoZIhvcNAQELBQAwgYYxCzAJBgNVBAYTAlhYMRIwEAYDVQQIDAlTdGF0ZU5hbWUxETAPBgNVBAcMCENpdHlOYW1lMRQwEgYDVQQKDAtDb21wYW55TmFtZTEbMBkGA1UECwwSQ29tcGFueVNlY3Rpb25OYW1lMR0wGwYDVQQDDBRDb21tb25OYW1lT3JIb3N0bmFtZTAeFw0yNDEyMTkwNjQ5MjRaFw0zNDEyMTcwNjQ5MjRaMIGGMQswCQYDVQQGEwJYWDESMBAGA1UECAwJU3RhdGVOYW1lMREwDwYDVQQHDAhDaXR5TmFtZTEUMBIGA1UECgwLQ29tcGFueU5hbWUxGzAZBgNVBAsMEkNvbXBhbnlTZWN0aW9uTmFtZTEdMBsGA1UEAwwUQ29tbW9uTmFtZU9ySG9zdG5hbWUwggIiMA0GCSqGSIb3DQEBAQUAA4ICDwAwggIKAoICAQCxSaia77e0mHOQLEW6hQyEft0Il1cejoLLGk3jUxNqW6G6OZT0OoJVUhErpnYmM3DU6z6c6Yv9qU7nN3TCkJtsCwJuKuyQY1nfTXOsFrgaDSF6cpsb3uion1u+ZhQ3DRW/VQcytwZd+kymeziIryiiFSQGSe+hjb+8masAViS5QhFQJDIGkiuG1WCTk9ov/EMDVsS8sdaWBIM82KpfC733JgTC5ObrxCvVuuB3RQGEDXzNGomO1HZUzOTJQI+OVG5ziWD9Oys2SPfs7IfNd7UfNwedix1zMiMQShfgS0wfQW80Lt9RH5WV9KgT3hh+J9edZd8yRdukSom0dr/4LEdWyrz20UZ7LWUFWnnjjmoc/Imti27MNcTNkmK4EexBaqFVHdN8TI+EU6SFmL6BVJUZGva/l21ZnO4DUnKAPV1l2WV79RlNyjavMEkxwR3BHlDUVogdMuf//uxjk8PUacpGH2UYBqlouPMW1qyq2WRV6IdUAbtRt/xKMe9FaFMY4ubFiYjNT5H6LxnysBboA0Eij6gQzrVzFRu10T72AMn0dfSZ6KEgaHMPZwQhJoa9eypW4oGESY43MtmHk3RtXoTDvic3EcivZVrRqydhvPIUnT/19DnpV4YkZ7u2rI+vRQ8/OZ7YfoIvGXuto3e9kZHVCghibUh9Ng40WKppVTcXxwIDAQABo1MwUTAdBgNVHQ4EFgQUSm5OwGtfZsSBvTvLF9lhNv3Pka4wHwYDVR0jBBgwFoAUSm5OwGtfZsSBvTvLF9lhNv3Pka4wDwYDVR0TAQH/BAUwAwEB/zANBgkqhkiG9w0BAQsFAAOCAgEAjGTX9QG0Z89ALBeuJaLfdQ/6FTUjTJCtt/aN4mVhHyxq5TL4rQMVb2wWtUUbvvmVktu31sc+YBerIIQ/n/1a5ZidsUBaKwXIgPHF5mIDV9/oSGSAxo1kX7PHy0kM42dEPnkSeoMabAeYV0F2cLkzm/Y1aX+8AX4R1G+2FcwMfLOfXVZi8rZXR1fkZ4Mr01rNseJJ60fJlBtOPV0ege9E4p27mLZLJd1xp07hHzJAfRM8P4Emkn4smLX0GhsB2H87s3canXWO3Wnh1nJgRmJ2SUhKf3WaWV8vMx9h4i2J73WKaf/6UUPw7LtOsEt11In0WUse27NcfEMeFzGBny24c6NC6kBZ0KUP+9p4hb3cKYzIWr/6tZ+Bz5V2Qvv00ryGw2o5frtMOAnJOG7rh0tO4nIJfy4tKgNMXXTERFikbuRVfurmFlibRzWDhLdRXq8FFgQuF7TXP4ljwwvwMHniGRkXdPwj6rXUBOiugCjlcui/eRvJCvITJjla4BIOwtXGIWveZ4BOPoooQlWte5ntnJuuXGYOkbozeP/P2LKM4pF4zKbdSbbrssJIc+drLUPe2K4WpChOHUDH8JUxZFHVAlzW2gKEOINwiI4yTCAFeiwZ/K8I8gyqb9hysd6dgsIDWkW+rXudoIyYy9o28QtXglKz2N/kkBKnGmW75geX/4E=" zoneid=5e86fe3c-ea4d-47ee-af4e-79095601ce1d
🙈 Error: (HTTP 530, error code 9999) The provided certificate name is invalid, please provide a valid name
```

Executed the api call 

https://cloudstack.apache.org/api/apidocs-4.18/apis/listTemplateDirectDownloadCertificates.html



(localcloud) 🐱 > list templatedirectdownloadcertificates listall=true


The failed certificates were not present

**Comment by kiranchavala** [comment]
@blueorangutan test

**Pull request merged**